TIGLCreator/src/TIGLCreatorSelectWingAndFlapStatusDialog.h

```diff
@@ -60,26 +61,35 @@ private slots:
     void on_checkSpoiler_stateChanged(int arg1);
 
 private:
-    Ui::TIGLCreatorSelectWingAndFlapStatusDialog *ui;
+    Ui::TIGLCreatorSelectWingAndFlapStatusDialog* ui;
 
     struct DeviceWidgets {
         QSlider* slider;
         QDoubleSpinBox* controlParamBox;
         QLabel* rotAngleLabel;
     };
 
-    std::map< std::string, DeviceWidgets> _guiMap;
-    std::map< std::string, tigl::CCPACSTrailingEdgeDevice*> _deviceMap;
+    struct RotationData {
+        double controlParam;
+        double rotation;
+        double factor;
+    };
+
+    std::map<std::string, DeviceWidgets> _guiMap;
+    std::map<std::string, tigl::CTiglAbstractGeometricComponent*> _deviceMap;
```

Comment:
The code works and reflects the previous implementation for TEDs. But I have a suggestion for refactoring. 

Problems: We store pointers to CTiglAbstractGeometricComponent in a map here. One conceptual issue is that not all CTiglAbstractGeometricComponents are control devices. Another issue is that we already manage a map that gives us geometric components given a uid. 

Solution: At all places where we currently use the `_deviceMap`, let's use the uid manager instead. We can query if a given uid is of an expected type, e.g. LED or TED and then resolve this object through the UID manager interface.